### ISSUE TITLE
systemd 256 fails seccomp on service startup due to sendto() usage

### ISSUE THREAD

**Issue by bobrik** [systemd/systemd#33299]
### systemd version the issue has been seen with

256

### Used distribution

Debian Bookworm

### CPU architectures issue was seen on

None

### Expected behaviour you didn't see

On systemd v255 we were able to start a unit without `sendto` allowed.

### Unexpected behaviour you saw

On systemd v256 the service fails to start:

```
Jun 12 22:01:22 131m33 systemd[1]: tracefwdr.service: Main process exited, code=exited, status=203/EXEC
```

We have the list of allowed syscalls in `SystemCallFilter=` and `SystemCallErrorNumber=EPERM` for this unit.

If I comment out `SystemCallErrorNumber=`, I can see a seccomp violation:

```
Jun 13 00:16:29 131m33 audisp-syslog[15587]: type=SECCOMP msg=audit(1718237789.318:96110): auid=4294967295 uid=2144 gid=2144 ses=4294967295 pid=1612773 comm="(racefwdr)" exe="/usr/lib/systemd/systemd-executor" sig=31 arch=c000003e syscall=44 compat=0 ip=0x7f8f1f529a39 code=0x80000000 AUID="unset" UID="tracefwdr" GID="tracefwdr" ARCH=x86_64 SYSCALL=unknown-syscall(-1)
```

Here 44 is `sendto`:

```
$ ausyscall 44
sendto
```

After some painful tracing due to `(` replacing the first character of the binary, one can find that `sendto()` is coming from inside the house:

```
(racefwdr) 1615623 [036] 15948.313137: probe:audit_log_format: (ffffffffa9c22180)
        ffffffffa9c22181 audit_log_format+0x1 ([kernel.kallsyms])
        ffffffffa9c2285d audit_log_start+0x49d ([kernel.kallsyms])
        ffffffffa9c43b77 audit_seccomp+0x57 ([kernel.kallsyms])
        ffffffffa9c6153e __seccomp_filter+0x99e ([kernel.kallsyms])
        ffffffffa9b4d390 syscall_trace_enter.isra.0+0x90 ([kernel.kallsyms])
        ffffffffabf441a5 do_syscall_64+0x15 ([kernel.kallsyms])
        ffffffffac000118 entry_SYSCALL_64_after_hwframe+0x60 ([kernel.kallsyms])
            7fcbe1329a39 __libc_send+0x19 (inlined)
            559571c7e9ae run+0x4de (/usr/lib/systemd/systemd-executor)
            559571c70396 main+0x36 (/usr/lib/systemd/systemd-executor)
            7fcbe1246249 __libc_start_call_main+0x79 (/usr/lib/x86_64-linux-gnu/libc.so.6)
```

It seems that there are some inlined frames that are missing, so I cannot pinpoint where exactly in the executor code the `sendto()` syscalls happens.

**Comment by bobrik** [comment]
I found seccomp violation coredumps and here's the stack:

```
(gdb) bt
#0  0x00007fc737b29a39 in __libc_send (fd=332, buf=buf@entry=0x7ffce66e3050, len=len@entry=16, flags=flags@entry=0) at ../sysdeps/unix/sysv/linux/send.c:28
#1  0x0000556be5937f02 in send_handoff_timestamp (reterr_exit_status=0x7ffce66e316c, p=0x7ffce66e3460, c=0x7ffce66e3570) at ../src/core/exec-invoke.c:4014
#2  exec_invoke (command=command@entry=0x7ffce66e3190, context=context@entry=0x7ffce66e3570, params=params@entry=0x7ffce66e3460, runtime=runtime@entry=0x7ffce66e3400,
    cgroup_context=cgroup_context@entry=0x7ffce66e3200, exit_status=exit_status@entry=0x7ffce66e316c) at ../src/core/exec-invoke.c:5330
#3  0x0000556be593b9af in run (argc=<optimized out>, argv=<optimized out>) at ../src/core/executor.c:236
#4  0x0000556be592d397 in main (argc=<optimized out>, argv=<optimized out>) at ../src/core/executor.c:267
```

It seems to come from #32441.

**Comment by bluca** [comment]
This is needed for tracking, so you need to allow it

**Comment by poettering** [comment]
Nah, *we* need to allow it, if the handoff stuff is used. THe syscalls we ourselves still need should be allowlisted by ourselves. THis is our bug.

**Comment by bluca** [comment]
Ah I misunderstood as an external filter

**Comment by yuwata** [comment]
Fix is waiting in #33451.
